### DISCUSSION TITLE
How to limit update rate of particular packages?

### DISCUSSION THREAD

**Discussion by kevincox** [renovatebot/renovate#28350] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

GitLab [config](https://gitlab.com/kevincox/renovate-runner/-/blob/master/.gitlab-ci.yml)

### Please tell us more about your question or problem

I am trying to limit the update rate of a package that frequently updates. Basically what I would like is to avoid updating the package again if it has been updated in the last N days.

I ended up with this config:

```json
{
	"packageRules": [{
		"groupName": "caniuse",
		"matchDatasources": ["npm"],
		"matchPackagePatterns": ["^caniuse(-|$)"],
		"extends": ["schedule:monthly"]
	}]
}
```

This mostly works but has a few problems.

1. If the package updates twice at "the wrong time" I can get 2 PRs on the same day.
2. Sometimes the package won't update.

This appears to be because the `schedule:monthly` preset [is defined as](https://docs.renovatebot.com/presets-schedule/#schedulemonthly) `before 4am on the first day of the month`. So if Renovate runs twice before 4am and there is a package update between those runs renovate will update twice. And if Renovate doesn't run during that time then no update will occur.

It seems that "schedule:monthly" is really just an approximation of once-a-month updates. Is there any option to get true once-a-month or whatever updates?

I imagine a solution like:

```jsonc
updateRateLimit: "30d",
```

Which would mean "skip updating this package/group if it was last updated within 30 days".

This could be potentially combined with `schedule` to get single updates in particular timeframes. For example `schedule:monthly` could potentially be redefined to

```json
{
	"schedule": "first day of the month",
	"updateRateLimit": "1d",
}
```

This would result in updates only occurring on the first day of the month and at most one time in that period.

**Comment by rarkins** [comment, marked as answer]
You might be able to achieve this indirectly using https://docs.renovatebot.com/configuration-options/#matchcurrentage

e.g.  matchPackageNames=x,y matchCurrentAge=<N days, enable=false

It will then disable updating the matching package names if their _current_ version is less than N days

**Reply by kevincox** [reply]
Thanks. That seems to fit my use case better. I haven't tested it but it sounds like it would also have edge cases, however I don't think those apply in my situation.

1. If the package updates twice, then just after N days you may get more frequent updates. For example if targeting update every 7 days:
    - day 0: v1 comes out
    - day 0: Update to v1
    - day 1: v2 comes out.
    - day 7: Update to v2
    - day 8: v3 comes out.
    - day 8: Update to v2 because v2 is now 7 days old.
2. I don't think it plays well with grouping as packages would just get dropped out of the group rather than result in the group getting disabled. But this is probably fine for many use cases.

However I think this would work perfectly for my use case as the package updates much more frequently than my target rate and I'm not grouping it. I'll try it out and post back with the results. (although that may take a few months)
